outputs.tf

```diff
@@ -1,27 +1,115 @@
-  this_public_ip                    = "${compact(concat(coalescelist(aws_instance.this.*.public_ip, aws_instance.this_t2.*.public_ip), list("")))}"
-  this_primary_network_interface_id = "${compact(concat(coalescelist(aws_instance.this.*.primary_network_interface_id, aws_instance.this_t2.*.primary_network_interface_id), list("")))}"
-  this_private_dns                  = "${compact(concat(coalescelist(aws_instance.this.*.private_dns, aws_instance.this_t2.*.private_dns), list("")))}"
-  this_private_ip                   = "${compact(concat(coalescelist(aws_instance.this.*.private_ip, aws_instance.this_t2.*.private_ip), list("")))}"
-  this_security_groups              = "${compact(concat(coalescelist(flatten(aws_instance.this.*.security_groups), flatten(aws_instance.this_t2.*.security_groups)), list("")))}"
-  this_vpc_security_group_ids       = "${compact(concat(coalescelist(flatten(aws_instance.this.*.vpc_security_group_ids), flatten(aws_instance.this_t2.*.vpc_security_group_ids)), list("")))}"
-  this_subnet_id                    = "${compact(concat(coalescelist(aws_instance.this.*.subnet_id, aws_instance.this_t2.*.subnet_id), list("")))}"
-  this_credit_specification         = "${aws_instance.this_t2.*.credit_specification}"
-  this_tags                         = "${coalescelist(flatten(aws_instance.this.*.tags), flatten(aws_instance.this_t2.*.tags))}"
+  this_id = compact(
+    concat(
+      coalescelist(aws_instance.this.*.id, aws_instance.this_t2.*.id),
+      [""],
+    ),
+  )
+  this_availability_zone = compact(
+    concat(
+      coalescelist(
+        aws_instance.this.*.availability_zone,
+        aws_instance.this_t2.*.availability_zone,
+      ),
+      [""],
+    ),
+  )
+  this_key_name = compact(
+    concat(
+      coalescelist(
+        aws_instance.this.*.key_name,
+        aws_instance.this_t2.*.key_name,
+      ),
+      [""],
+    ),
+  )
+  this_public_dns = compact(
+    concat(
+      coalescelist(
+        aws_instance.this.*.public_dns,
+        aws_instance.this_t2.*.public_dns,
+      ),
+      [""],
+    ),
+  )
+  this_public_ip = compact(
+    concat(
+      coalescelist(
+        aws_instance.this.*.public_ip,
+        aws_instance.this_t2.*.public_ip,
+      ),
+      [""],
+    ),
+  )
+  this_primary_network_interface_id = compact(
+    concat(
+      coalescelist(
+        aws_instance.this.*.primary_network_interface_id,
+        aws_instance.this_t2.*.primary_network_interface_id,
+      ),
+      [""],
+    ),
+  )
+  this_private_dns = compact(
+    concat(
+      coalescelist(
+        aws_instance.this.*.private_dns,
+        aws_instance.this_t2.*.private_dns,
+      ),
+      [""],
+    ),
+  )
+  this_private_ip = compact(
+    concat(
+      coalescelist(
+        aws_instance.this.*.private_ip,
+        aws_instance.this_t2.*.private_ip,
+      ),
+      [""],
+    ),
+  )
+  this_security_groups = compact(
```

Comment:
due to the modification of the functionality of `coalescelist` this will break until 0.12.1 if no `security_group` is passed.

see https://github.com/hashicorp/terraform/issues/21370

temporary workaround is to include a bogus value in the last array instead of an empty string